src/py/flwr/server/state/sqlite_state.py

```diff
@@ -36,6 +37,12 @@
 );
 """
 
+SQL_CREATE_TABLE_WORKLOAD = """
+CREATE TABLE IF NOT EXISTS workload(
+    workload_id TEXT UNIQUE
```

Comment:
`SQL_CREATE_TABLE_TASK_INS` should be modified to make workload_id a foreign key to check the constraint automatically.

Reply:
sure, makes a lot of sense. will change this.